diff --git a/scripts/firecracker.py b/scripts/firecracker.py
index d0b24071..b26f6cd7 100755
--- a/scripts/firecracker.py
+++ b/scripts/firecracker.py
@@ -237,7 +237,7 @@ def start_firecracker_with_no_api(firecracker_path, firecracker_config_json):
     api_file.write(bytes(firecracker_config_json, 'utf-8'))
     api_file.flush()
     stty_save()
-    return subprocess.Popen([firecracker_path, "--no-api", "--config-file", api_file.name],
+    return subprocess.Popen(['/usr/bin/time', '-f', 'Firecracker maxRSS: %M', firecracker_path, "--no-api", "--config-file", api_file.name],
                            stdout=sys.stdout, stderr=subprocess.STDOUT), api_file.name
 
 
